src/control/dmg/pool.go

```diff
@@ -23,7 +23,177 @@
 
 package main
 
-// TODO: provide implementation to this placeholder
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/daos-stack/daos/src/control/common"
+	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/inhies/go-bytesize"
+	"github.com/pkg/errors"
+)
+
+const (
+	msgSizeNoNumber = "size string doesn't specify a number"
+	msgSizeBad      = "size string illegal, %s"
+	msgSizeZeroScm  = "non-zero scm size is required"
+	maxNumSvcReps   = 13
+)
 
 // PoolCmd is the struct representing the top-level pool subcommand.
-type PoolCmd struct{}
+type PoolCmd struct {
+	Create CreatePoolCmd `command:"create" alias:"c" description:"Create a DAOS pool"`
+}
+
+// CreatePoolCmd is the struct representing the command to create a DAOS pool.
+type CreatePoolCmd struct {
+	GroupName  string `short:"g" long:"group" description:"DAOS pool to be owned by given group"`
+	UserName   string `short:"u" long:"user" description:"DAOS pool to be owned by given user"`
+	ACLFile    string `short:"a" long:"acl-file" description:"Access Control List file path for DAOS pool"`
+	ScmSize    string `short:"s" long:"scm-size" required:"1" description:"Size of SCM component of DAOS pool"`
+	NVMeSize   string `short:"n" long:"nvme-size" description:"Size of NVMe component of DAOS pool"`
+	RankList   string `short:"r" long:"ranks" description:"DAOS pool to be owned by given user"`
+	NumSvcReps uint32 `short:"v" long:"nsvc" default:"1" description:"Number of pool service replicas"`
+	//GID uint32 `short:"g" long:"gid" description:"Pool GID"`
+	//UID uint32 `short:"u" long:"uid" description:"Pool UID"`
+	//Group string `short:"G" long:"group" description:"Pool group name"`
+	//Mode uint64 `short:"m" long:"mode" description:"Pool permissions mode"`
```

Comment:
Should remove the commented out params no longer used

Reply:
thanks was waiting for confirmation of the changes but looks like I now have that.